pages/editTrip/[id].js

```diff
@@ -0,0 +1,140 @@
+import ConfirmDelete from "@/components/ConfirmDelete";
+import ConfirmationMessage from "@/components/ConfirmationMessage";
+import Link from "next/link";
+import { useRouter } from "next/router";
+import useSWR from "swr";
+
+const fetcher = (...args) => fetch(...args).then((res) => res.json());
+
+export default function EditTrip() {
+  const { mutate } = useSWR("/api/trips");
+  const router = useRouter();
+  const { id } = router.query;
+  const { data: trips, isLoading } = useSWR(
+    id ? `/api/trips/${id}` : null,
+    fetcher
+  );
+
+  if (!trips || isLoading) {
+    return <h2>is Loading...</h2>;
+  }
+
+  async function handleEdit(event) {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const tripData = Object.fromEntries(formData);
+    const response = await fetch(`/api/trips/${id}`, {
+      method: "PUT",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(tripData),
+    });
```

Comment:
It's entirely fine to have the fetch operations directly in the handler functions ^^

Since you'll have multiple of them, it's also good to move them to a `src/lib/api.js` file ([remember this](https://github.com/neuefische/web-exercises/blob/crud-recap/examples/crud-recap/lib/api.js))

Reply:
Good idea, we' re going to refactor these code-ideas in our User Story 10 – Refactor One.